src/librustdoc/html/render/search_index.rs

```diff
@@ -306,19 +301,17 @@ fn add_generics_and_bounds_as_types<'tcx>(
                 return;
             }
         }
-        let mut index_ty = get_index_type(&ty, generics);
-        if index_ty.name.as_ref().map(|s| s.is_empty()).unwrap_or(true) {
-            return;
-        }
+        let Some(mut index_ty) = get_index_type(&ty, generics)
+        else { return };
         if is_full_generic {
             // We remove the name of the full generic because we have no use for it.
-            index_ty.name = Some(String::new());
```

Comment:
@camelid

For the record, I approve of this refactoring. If the generated JSON uses the empty string as its sentinel here (and it should, because it's smaller), then RenderType should just use the empty string here to match.

Reply:
There is another open PR which fixes this part but can't find it... T_T